### ISSUE TITLE
L* Gamma support?

### ISSUE THREAD

**Issue by radioproektor** [cameramanben/LUTCalc#8]
Is it possible to add L* Gamma to your app? 
Here are some links about it:
http://stonerosephotos.com/blog/2010/09/gamma-the-zone-system/
http://forum.luminous-landscape.com/index.php?topic=38535
https://www.dpreview.com/forums/thread/3069106
http://www.colormanagement.org/en/workingspaces.html

**Comment by radioproektor** [comment]
one more useful link http://www.eye-q.de/uploads/media/LStar-Dokumentation.pdf

**Comment by radioproektor** [comment]
Also you can find refence ICC v4 profiles with parametric L* gamma curve here: https://github.com/ellelstone/elles_icc_profiles
The  L* gamma formula looks like this:
![screen shot 2018-01-05 at 9 27 00 am](https://user-images.githubusercontent.com/35053925/34599384-b050d6e8-f1fa-11e7-8c8c-733f3b4ba56b.png)

**Comment by radioproektor** [comment]
"A Continuity Study of the CIE L* Function by Bruce J. Lindbloom" http://www.brucelindbloom.com/LContinuity.html

**Comment by radioproektor** [comment]
Sorry for bumping this topic again but any updates here? Is this project alive? 
L* Gamma is widely used by many photographers, it is also used for film scans processing and editing because allows large adjustments without boosting contrast relations.
linear to  L* Gamma and L* Gamma to g2.2 LUTs can make real magic for image editing.

**Comment by Entropy512** [comment]
I haven't seen Ben make any commits in quite some time, I don't know if it's because he's busy with other stuff or considers things to be mostly complete.

You could add the transfer curve with the properties you desire in gamma.js and submit a pull request?

**Comment by radioproektor** [comment]
I am not a coder and i don't understand how to build that transfer curve. I found another source with a lot math formulas and explanation of this gamma http://www.marcelpatek.com/gamma.html search for L-star curve part in the middle of the article.

**Comment by radioproektor** [comment]
So any hope to see this option somehow?
L* gamma curve is perfect for inverting film negatives. Here is real life usage example: for most accurate results film negative scanned in linear gamma and should be inverted before transformation to working color space. Invert tool don't works in linear gamma as it should, so we need to transform Linear to L* gamma, next do invert, and next transform back from L* gamma to Linear.
If instead L*gamma we use sRGB or something like 2.2gamma, invert will produce less accurate results and will shift tonal and color values after invert.

**Comment by radioproektor** [comment]
some additional info and formulas here https://en.wikipedia.org/wiki/Lightness

**Comment by radioproektor** [comment]
Great to see this project is back to life. Any hope to see some progress with L* gamma?
By the way, i noticed that L* gamma seems may be different between apps. For Example ICC v4 profile generated with ICC Profile Editor in RawTherapee have some additional lines in formula. As result if i use v4 ICC profile in PhotoLine app, some correction tools don't work as expected. Seems some compatibility issues with ICC v4.
![Screen Shot 2020-11-05 at 9 51 41 PM](https://user-images.githubusercontent.com/35053925/98290047-bd44b080-1fb1-11eb-803a-891c5460dc0c.jpg)

**Comment by cameramanben** [comment]
Hi radioproektor,

I've just uploaded code to add CIE L* to the gamma list (it's with the other power function and linear options).

Just for fun, I've also added CIELAB as a gamut option. Considering that it is rather less useful than the L* gamma in a LUT, it is a bit complicated so that it fits with the whole 'combine CIELAB gamut with L* gamma to make CIELAB colourspace' workflow of LUTCalc. Still, it looks like it is working (ie it looks very odd in an RGB based system!) ;-)

It's in the main branch now, and I have it up for testing at [lutcalc.net/testing](https://lutcalc.net/testing)

Ben
